test/e2e/metrics_test.go

```diff
@@ -378,3 +398,11 @@ func suffixize(name string, idx int) string {
 
 	return fmt.Sprintf("%s-%d", name, idx)
 }
+
+func getPodNameByLabel(namespace, labelKey, labelValue string) string {
```

Comment:
I don't like that this helper is defined in this file, but shared between the tracing and metric test. Also the assumption that it has only one pod is very specific. I'm wondering if we should rather move fetching the pod into the `Eventually` block of the test cases and improve the robustness by that while accepting duplicating these few lines of code.